### PR TITLE
Rename duplicate file name

### PR DESCRIPTION
**Pull request by chaoskao**
What I did:
Rename duplicate file name.

Why I did it:
It it hard to checkout code to non-case sensitive system.

How I verified it:
Use git clone repo to Windows or OSX and code build success by mount to the docker.

### CODE REVIEW

**Comment by kcudnik** [comment]
retest vs please

File: syncd/Makefile.am [hunk 1]
```diff
@@ -50,7 +50,7 @@ libSyncd_a_SOURCES = \
 				SaiObj.cpp \
 				SaiAttr.cpp \
 				VendorSai.cpp \
-				syncd.cpp \
+				_syncd.cpp \
```
Comment (kcudnik) [inline syncd/Makefile.am:53]: rename to syncd_main.cpp

File: saiplayer/Makefile.am [hunk 2]
```diff
@@ -17,6 +17,6 @@ libSaiPlayer_a_SOURCES = \
 
 libSaiPlayer_a_CPPFLAGS = $(DBGFLAGS) $(AM_CPPFLAGS) $(CFLAGS_COMMON) -std=c++14
 
-saiplayer_SOURCES = saiplayer.cpp
+saiplayer_SOURCES = _saiplayer.cpp
```
Comment (kcudnik) [inline saiplayer/Makefile.am:20]: rename to saiplayer_main.cpp

**Comment by kcudnik** [comment]
saiattributelist.h i used in swss orchagent, without this it will probably fail

**Review by kcudnik — COMMENTED** [review]
please rename files in some fashion not to use "_" as starting file name

**Review by kcudnik — CHANGES_REQUESTED** [review]
please rename files in some fashion not to use "_" as starting file name

**Comment by kcudnik** [comment]
are you planning to build entire sonic image on windows/osx?

File: syncd/Makefile.am (hunk 1 above)
Comment (chaoskao) [reply syncd/Makefile.am:53]: ok

**Comment by chaoskao** [comment]
I check the saiattributelist.h only used in mock_tests/check.h, others used the SaiAttributeList.h. Maybe we can rename to saiattributelist_mock.h?
I am not planning to build sonic-buildimage on Windows/OSX, just git clone to Windows/OSX and mount to docker do some test.

**Comment by kcudnik** [comment]
> I check the saiattributelist.h only used in mock_tests/check.h, others used the SaiAttributeList.h. Maybe we can rename to saiattributelist_mock.h?

this is used in sonic-swss repo, and i think it needs to be fixed there

**Comment by chaoskao** [comment]
Hi @kcudnik you can reference this [branch ](https://github.com/chaoskao/sonic-swss/tree/legacy_code) for your PR [Azure/sonic-swss#1608](https://github.com/Azure/sonic-swss/pull/1608)

**Comment by kcudnik** [comment]
what do you mean?

**Comment by kcudnik** [comment]
swss-common was merged, and you can remove that saiattributelist.h file completely

**Review by kcudnik — APPROVED** [review]

**Pull request merged**